sotodlib/core/flagman.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+from functools import reduce
+
+from so3g.proj import Ranges, RangesMatrix
+from . import AxisManager
+
+class FlagManager(AxisManager):
+    """An extension of the AxisManager class to make functions 
+    more specifically associated with cuts and flags.
+    
+    FlagManager only expects to have items that are mapped to the 
+    detector axis, the sample axis, or both. 
+    
+    Detector Flags can be passed as bitmasks or boolean arrays. To match with
+    Ranges and RangesMatrix, the default is False and the exceptions
+    are True
+    """
+    
+    def __init__(self, det_axis, samp_axis):
+        super().__init__(det_axis, samp_axis)
+                
+        if not 'dets' in self._axes:
+            raise ValueError('FlagManagers require a dets axis')
+        if not 'samps' in self._axes:
+            raise ValueError('FlagManagers require a samps axis')
+            
+    def wrap(self, name, data, axis_map=None, *kwargs):
+        """If axis_map is None, the data better be (dets,), (samps,),
+            or (dets, samps)
+        """
+        
+        if axis_map is None:
+            try:
+                s = data.shape
+            except:
+                ### catches if a detector mask is just a list
+                s = np.shape(data)
+            if len(s) == 1:
+                if s[0] == self.dets.count:
+                    ## detector only flag. Turn into RangesMatrix
+                    axis_map=[(0,'dets')]
+                elif s[0] == self.samps.count:
+                    axis_map=[(0,'samps')]
```

Comment:
I'm going to do a couple things here:

1) I'll raise a "cannot auto determine type" error if `dets.count==samps.count`, that will deal with that possible edge case. 
2) We're going to enforce the dets/samps ordering for 2D flags. Because I realize other parts of this code assumes that exists anyway.

Reply:
And 3. Going to define `wrap_samps`, `wrap_dets`, `wrap_dets_samps` too, since if you're building a pipeline type thing those could be used by default.